cdc panic on the upstream manager

### What did you do?

run the changefeed normally

### What did you expect to see?

nothing

### What did you see instead?

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2d154b7]

goroutine 135 [running]:
github.com/pingcap/tiflow/pkg/upstream.(*Manager).Tick(0xc000ba2fc0, {0x7f277863e980, 0xc00137c0a0}, 0xc000b89260)
	github.com/pingcap/tiflow/pkg/upstream/manager.go:200 +0x1f7
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick(0xc0008c6d40, {0x7f277863e980?, 0xc00137c0a0}, {0x4307890?, 0xc000b89260})
	github.com/pingcap/tiflow/cdc/processor/manager.go:149 +0x585
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc002ec49a0, {0x7f277863e980, 0xc00137c0a0}, 0xc000b9c480, 0x2faf080, {0x39e726d, 0x9})
	github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290 +0xfa6
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc0018f5560, {0x432f290?, 0xc00137c0a0}, {0x7f2778a5d3a0, 0xc0008c6d40}, {0x4307890, 0xc000b89260}, 0xc002ea1788?, {0x39e726d, 0x9})
	github.com/pingcap/tiflow/cdc/capture/capture.go:550 +0x185
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func4()
	github.com/pingcap/tiflow/cdc/capture/capture.go:382 +0x3a5
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2d154b7]
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
v6.5.5
```

--- a/pkg/upstream/manager.go
+++ b/pkg/upstream/manager.go
@@ -196,7 +196,9 @@ func (m *Manager) Tick(ctx context.Context,
 
 	activeUpstreams := make(map[uint64]struct{})
 	for _, cf := range globalState.Changefeeds {
-		activeUpstreams[cf.Info.UpstreamID] = struct{}{}
+		if cf != nil && cf.Info != nil {
+			activeUpstreams[cf.Info.UpstreamID] = struct{}{}
+		}
 	}
 	m.mu.Lock()
 	defer m.mu.Unlock()

